components/calculator.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Buttons from './buttons';
+import Display from './display';
+import calculate from '../logic/calculate';
+
+// eslint-disable-next-line react/prefer-stateless-function
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    const {
+      previous,
+      operand,
+      current,
+    } = props;
+    this.state = {
+      total: previous,
+      operation: operand,
+      next: current,
+    };
+    this.update = this.update.bind(this);
+  }
+
+  update(btnClick) {
+    const obj = calculate(this.state, btnClick);
+    this.setState(() => ({
+      total: obj.total,
+      operation: obj.operation,
+      next: obj.next,
+    }));
+  }
+
+  render() {
+    const { total, operation, next } = this.state;
+    return (
+      <div className="container">
```

Comment:
- [ ] You should use `useState` to handle state. You can follow this [link ](https://www.w3schools.com/react/react_usestate.asp)to know how to use it. kindly work on it.

Reply:
@uwadonat  


Uses state was not stated ro be mandatory, a mere sub note